osmomath/decimal.go

```diff
@@ -680,6 +695,29 @@ func chopPrecisionAndRoundUp(d *big.Int, precisionReuse *big.Int) *big.Int {
 	return quo.Add(quo, oneInt)
 }
 
+// chopPrecisionAndRoundUp removes a Precision amount of rightmost digits and rounds up.
+// mutative input
+func chopPrecisionAndRoundUpMut(d *big.Int, precisionReuse *big.Int) *big.Int {
+	// remove the negative and add it back when returning
+	if d.Sign() == -1 {
+		// make d positive, compute chopped value, and then un-mutate d
+		d = d.Neg(d)
+		// truncate since d is negative...
+		d = chopPrecisionAndTruncateMut(d)
+		d = d.Neg(d)
+		return d
+	}
+
+	// get the truncated quotient and remainder
+	_, rem := d.QuoRem(d, precisionReuse, big.NewInt(0))
+
+	if rem.Sign() == 0 { // remainder is zero
+		return d
+	}
+
+	return d.Add(d, oneInt)
+}
+
```

Comment:
How does this differ from `chopPrecisionAndRoundUp`? I'm not seeing any difference

Reply:
I'm using `chopPrecisionAndTruncateMut` that uses `d` as result. Also use `d` as `QuoRem` res instead of declaring new `quo` variable

Reply:
Thanks for the context. I've made commits directly to the branch with changes that were missing from approval.

What was added:
- remove the non-mutative chop version since we can now use the mutative one + copy the input 
- covered non-mutative QuoRoundUp with a test

Commit: [0622c30](https://github.com/osmosis-labs/osmosis/pull/6437/commits/0622c30178eb1c4287c6dac69d7d884603e7487e)